src/payment/strategies/moneris/moneris-payment-strategy.ts

```diff
@@ -0,0 +1,147 @@
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+
+        const nonce = await new Promise<string | undefined>((resolve, reject) => {
+            if (!this._iframe) {
+                throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+            }
+
+            const frameref = this._iframe.contentWindow;
+
+            frameref?.postMessage('tokenize', this._monerisURL(!!testMode));
+
+            this._windowEventListener = (response: MessageEvent) => this._handleMonerisResponse(response, resolve, reject);
+
+            window.addEventListener('message', this._windowEventListener);
+        });
+
+        if (nonce !== undefined) {
+            return this._store.dispatch(this._paymentActionCreator.submitPayment({
+                methodId: payment.methodId,
+                paymentData: { nonce },
+            }));
+        }
+
+        return this._store.getState();
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        if (this._windowEventListener) {
+            window.removeEventListener('message', this._windowEventListener);
+            this._windowEventListener = undefined;
+        }
+
+        if (this._iframe && this._iframe.parentNode) {
+            this._iframe.parentNode.removeChild(this._iframe);
+            this._iframe = undefined;
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _createIframe(containerId: string, profileId: string, testMode: boolean, style?: MonerisStylingProps): HTMLIFrameElement {
+        const container = document.getElementById(containerId);
+
+        if (!container) {
+            throw new InvalidArgumentError('Unable to create iframe without valid container ID.');
+        }
+
+        const iframe = document.createElement('iframe');
+        // Example CSS styling as provided by Moneris' documentation
+        const cssBody = style?.cssBody ?? 'background:white;';
+        const cssTextbox = style?.cssTextbox ?? 'border-width:2px;';
+        const csstextboxPan = style?.cssTextboxPan ?? 'width:140px;';
+        const cssTextboxExpiry = style?.cssTextboxExpiry ?? 'width:40px;';
+        const csstexboxCvd = style?.csstexboxCvd ?? 'width:40px';
+
+        iframe.name = IFRAME_NAME;
+        iframe.id = IFRAME_NAME;
+
+        iframe.src = `${ this._monerisURL(testMode) }?id=${profileId}&pmmsg=true&css_body=${cssBody}&css_textbox=${cssTextbox}&css_textbox_pan=${csstextboxPan}&enable_exp=1&css_textbox_exp=${cssTextboxExpiry}&enable_cvd=1&css_textbox_cvd=${csstexboxCvd}&display_labels=1`;
+
+        container.appendChild(iframe);
+
+        return iframe;
+    }
+
+    private _handleMonerisResponse(response: MessageEvent, resolve: (nonce: string) => void, reject: (reason: Error) => void): void {
```

Comment:
It is a bit awkward to pass the responsibility of resolving/rejecting the promise into this function. Might be better to keep this function just about handling the response and move the `resolve`/`reject` part to the caller.

```ts
private _handleMonerisResponse(response: MessageEvent): string {
```

```ts
this._windowEventListener = (response: MessageEvent) => {
  try {
    resolve(this._handleMonerisResponse(response));
  } catch (error) {
    reject(error);
  }
};
```